app/services/builders/external_pull_request.rb

```diff
@@ -0,0 +1,26 @@
+module Builders
+  class ExternalPullRequest < BaseService
+    def initialize(pull_request_data, external_project)
+      @pull_request_data = pull_request_data
+      @external_project = external_project
+    end
+
+    def call
+      ::ExternalPullRequest.find_or_create_by!(github_id: @pull_request_data[:id]) do |pull_request|
+        pull_request.html_url = @pull_request_data[:html_url]
+        pull_request.body = @pull_request_data[:body]
+        pull_request.title = @pull_request_data[:title]
+        pull_request.owner_id = User.find_by!(login: @pull_request_data[:user][:login]).id
+        pull_request.external_project_id = built_or_saved_project.id
+      end
+    end
+
+    private
+
+    def built_or_saved_project
+      return @external_project if @external_project.persisted?
+
+      @external_project.tap(&:save!)
```

Comment:
why is that we don't save the record if it was already persisted? we don't want to update the attributes of the external PR?

Reply:
yes, we are not tracking almost anything about the external_project, in fact we only use the github_id just to get all the pull request and for pull requests we only want the html_url attribute, that's why i did it like that, im not sure if changes like full name or name will change things for us when processing.. besides that we want to avoid unnecessary queries to the DB because of the number of projects o pull request that a member could have